substate-store-impls/src/hash_tree/substate_tier.rs

```diff
@@ -213,16 +213,31 @@ impl<'s, S: ReadableTreeStore + WriteableTreeStore> SubstateTier<'s, S> {
         for (key, node) in tree_update_batch.node_batch.iter().flatten() {
             // We promised to associate Substate values; but not all newly-created nodes are leaves:
             if let Node::Leaf(leaf_node) = &node {
-                // And not every newly-created leaf comes from a value change: (sometimes it is just a tree re-structuring!)
                 let sort_key = Self::to_typed_key(leaf_node.leaf_key().clone());
-                if let Some(substate_value) = substate_updates.get_upserted_value(&sort_key) {
-                    self.base_store.associate_substate(
-                        &self.stored_node_key(&key),
-                        &self.partition_key,
-                        &sort_key,
-                        substate_value,
-                    );
-                }
+                let substate_value = match substate_updates {
```

Comment:
I quite liked the `get_upserted_value` abstraction... what do you think about getting it back, but perhaps having it be slightly more detailed. E.g. be `get_substate_change` and return:
```
enum SubstateChange {
    Upsert(..), // Delta>Set or Reset>Present
    Delete,  // Delta>Delete or Reset>NotPresent
    None, // Delta>NotPresent
}
```
Which we can then handle more tersely here.

Reply:
I agree, it is useful 👍  (I just allowed myself to express `None` via regular `Option`)